engine/src/main/java/org/terasology/world/generation/facets/base/BaseFacet2D.java

```diff
@@ -18,43 +18,46 @@
 import org.joml.Vector3i;
 import org.terasology.math.geom.Rect2i;
 import org.terasology.math.Region3i;
+import org.terasology.world.block.BlockArea;
+import org.terasology.world.block.BlockAreac;
 import org.terasology.world.block.BlockRegion;
+import org.terasology.world.block.BlockRegionc;
 import org.terasology.world.generation.Border3D;
 import org.terasology.world.generation.WorldFacet2D;
 
 /**
  */
 public class BaseFacet2D implements WorldFacet2D {
 
-    private Rect2i worldRegion;
-    private Rect2i relativeRegion;
+    private BlockArea worldRegion;
+    private BlockArea relativeRegion;
 
-    public BaseFacet2D(BlockRegion targetRegion, Border3D border) {
+    public BaseFacet2D(BlockRegionc targetRegion, Border3D border) {
```

Comment:
We should definitely add some JavaDoc here stating how the 3D block region `targetRegion` is translated into a 2D area. In particular, how the projection down to 2D works...

Reply:
we could always do with more javadocs